File: cpp/7_整数反转_easy.cpp
class Solution {
public:
    int reverse(int x) {
        if (!x) return 0;
        int y = x;
        int ans = 0;
        while (y) {
            if (ans < INT_MIN/10 || ans > INT_MAX/10)
                return 0;
            ans = ans*10 + y%10;
            y /= 10;
        }
        return ans;
    }
};

File: python/7_整数反转_easy.py
class Solution:
    def reverse(self, x: int) -> int:
        if not x:
            return x
        INT_MIN, INT_MAX = -1<<31, (1<<31)-1
        ans = 0
        while x:
            if ans < INT_MIN//10 + 1 or ans > INT_MAX//10:
                return 0
            digit = x % 10
            if x < 0 and digit > 0:
                digit -= 10
            ans = ans*10 + digit
            x = (x-digit)//10
        return ans